Sources/Misc/Purchases+nonasync.swift

```diff
@@ -0,0 +1,109 @@
+     * - Parameter completion: A completion block that is called when the ``RefundRequestStatus`` is returned.
+     * Keep in mind the status could be ``RefundRequestStatus/userCancelled``
+     * If the request was unsuccessful, no active entitlements could be found for the user,
+     * or multiple active entitlements were found for the user, an `Error` will be thrown.
+     */
+    @available(iOS 15.0, *)
+    @available(macOS, unavailable)
+    @available(watchOS, unavailable)
+    @available(tvOS, unavailable)
+    func beginRefundRequest(
+        forProduct productID: String,
+        completion: @escaping (Result<RefundRequestStatus, Error>) -> Void
+    ) {
+        call(completion) {
+            try await self.beginRefundRequest(forProduct: productID)
+        }
+    }
+
+    /**
+     * Presents a refund request sheet in the current window scene for
+     * the latest transaction associated with the entitlement ID.
+     *
+     * - Parameter entitlementID: The entitlementID to begin a refund request for.
+     * - Parameter completion: A completion block that is called when the ``RefundRequestStatus`` is returned.
+     * Keep in mind the status could be ``RefundRequestStatus/userCancelled``
+     * If the request was unsuccessful, no active entitlements could be found for the user,
+     * or multiple active entitlements were found for the user, an `Error` will be thrown.
+     */
+    @available(iOS 15.0, *)
+    @available(macOS, unavailable)
+    @available(watchOS, unavailable)
+    @available(tvOS, unavailable)
+    func beginRefundRequest(
+        forEntitlement entitlementID: String,
+        completion: @escaping (Result<RefundRequestStatus, Error>) -> Void
+    ) {
+        call(completion) {
+            try await self.beginRefundRequest(forEntitlement: entitlementID)
+        }
+    }
+
+    /**
+     * Presents a refund request sheet in the current window scene for
+     * the latest transaction associated with the active entitlement.
+     *
+     * - Parameter completion: A completion block that is called when the ``RefundRequestStatus`` is returned.
+     * Keep in mind the status could be ``RefundRequestStatus/userCancelled``
+     * If the request was unsuccessful, no active entitlements could be found for the user,
+     * or multiple active entitlements were found for the user, an `Error` will be thrown.
+     *
+     * - Important: This method should only be used if your user can only
+     * have a single active entitlement at a given time.
+     * If a user could have more than one entitlement at a time, use ``beginRefundRequest(forEntitlement:)`` instead.
+     */
+    @available(iOS 15.0, *)
+    @available(macOS, unavailable)
+    @available(watchOS, unavailable)
+    @available(tvOS, unavailable)
+    func beginRefundRequestForActiveEntitlement(
+        completion: @escaping (Result<RefundRequestStatus, Error>) -> Void
+    ) {
+        call(completion) {
```

Comment:
It feels a bit weird to `call` the completion before the actual block... Maybe we could rename to `awaitWithCompletion` or something like that?

Reply:
Yeah this was a bad name, since it was a private function I didn't think too hard about it hehe, but yeah that's a better name I'll change it,.

Reply:
Yeah this was a bad name, since it was a private function I didn't think too hard about it hehe, but yeah that's a better name I'll change it.

Reply:
Actually the Swifty version would be `await(with completion:)`, but we can't make a function named `await`. I'll rename this to `call(with completion:)`